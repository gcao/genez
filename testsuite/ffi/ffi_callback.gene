# Test FFI callbacks (simplified example)

# For now, we'll simulate a callback scenario
# In a full implementation, this would actually pass Gene functions to C

# Define a Gene function that could be used as a callback
(fn double [x]
  (* x 2))

# Test the callback creation syntax
(var cb (c-callback double "int(int)"))
(print "Created callback")

# For now, just show we can create arrays with the doubled values
(var test_array [1 2 3 4 5])
(print "Original: " test_array)

# Manually create the doubled array since ++ is not implemented
(var result [2 4 6 8 10])
(print "Doubled: " result)