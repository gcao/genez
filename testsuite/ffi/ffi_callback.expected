Original:  [1, 2, 3, 4, 5]
Doubled:  [2, 4, 6, 8, 10]